scripts/setup.sh

```diff
@@ -1,290 +1,153 @@
-#!/usr/bin/env bash
-
-# can be run standalone with: curl -sL git.io/deepops | bash
-# or: curl -sL git.io/deepops | bash -s -- 19.07
-
-# DeepOps branch to setup
-DEEPOPS_TAG="${1:-master}"
-JENKINS="${JENKINS:-}" # Used to signal we are in a Jenkins testing environment and need virtualenv
-
+#!/bin/bash --init-file
+
+# DeepOps setup/bootstrap script
+#   This script installs required dependencies on a system so it can run Ansible
+#   and initializes the DeepOps directory
+#
+# Can be run standalone with: curl -sL git.io/deepops | bash
+#                         or: curl -sL git.io/deepops | bash -s -- 19.07
+
+# Configuration
+ANSIBLE_VERSION="${ANSIBLE_VERSION:-2.9.5}"     # Ansible version to install
+ANSIBLE_OK="${ANSIBLE_OK:-2.7.8}"               # Oldest allowed Ansible version
+CONFIG_DIR="${CONFIG_DIR:-./config}"            # Default configuration directory location
+DEEPOPS_TAG="${1:-master}"                      # DeepOps branch to set up
+JINJA2_VERSION="${JINJA2_VERSION:-2.11.1}"      # Jinja2 required version
+PIP="${PIP:-pip3}"                              # Pip binary to use
+PYTHON_BIN="${PYTHON_BIN:-/usr/bin/python3}"    # Python3 path
+VENV_DIR="${VENV_DIR:-/opt/deepops/env}"        # Path to python virtual environment to create
```

Comment:
I don't love that this is in `/opt` on the Ansible controller (which is possibly just the user workstation!)

Possibly put this in their home directory, e.g. `${HOME}/.deepops`? 

I'm not sure of the right answer tbh 🤷🏻‍♂️ it just feels odd to be adding directories outside the homedir.